File: skill_sdk/tests/fixtures/js-project/src/core/DataProcessor.js
/**
 * Defines the contract for processing data.
 * Analogous to the DataProcessor interface in the Java fixture.
 *
 * @interface DataProcessor
 */
export class DataProcessor {
  /**
   * Process raw data and return the processed result.
   * @param {string} data
   * @returns {string}
   */
  process(data) {
    throw new Error('Not implemented');
  }

  /**
   * Validate whether the given data can be processed.
   * @param {string} data
   * @returns {boolean}
   */
  validate(data) {
    throw new Error('Not implemented');
  }
}

File: skill_sdk/tests/fixtures/rust-project/src/core/data_processor.rs
/// DataProcessor trait defines the contract for data processing.
///
/// goToDefinition on DataProcessor should resolve here.
/// findReferences on DataProcessor should find all usages across core/ and shop/.
/// goToImplementation on DataProcessor should resolve to DefaultProcessor.
pub trait DataProcessor: Send + Sync {
    /// Process applies the data processing pipeline.
    /// hover on Process should show this documentation comment.
    fn process(&self, data: &str) -> Result<String, String>;

    /// Validate checks whether the provided data is valid.
    fn validate(&self, data: &str) -> bool;
}
